tests/unit/test_git_hooks.py

```
@@ -0,0 +1,288 @@
+
+    def test_guard_checks_renamed_tracking_branch(self, tmp_path):
+        """A branch tracking a NON-default remote ref (renamed) is NOT base-
+        tracking — it has its own remote branch and must still be checked.
+
+        Before the fix, any upstream-name != local-name branch was skipped, so
+        a merged renamed branch would wrongly accept commits.
+        """
+        import subprocess
+
+        remote = tmp_path / "remote.git"
+        subprocess.run(["git", "init", "-q", "--bare", str(remote)], check=True)
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)